src/main.rs

```diff
@@ -127,53 +166,97 @@ async fn run() -> Result<(), Box<dyn std::error::Error>> {
     }
 
     if config.id.is_none() || config.ids.is_none() {
-        error!("Failed to find required identites, try running with `-g`");
+        error!(
+            root_logger,
+            "Failed to find required identites, try running with `-g`"
+        );
         return Ok(());
     }
 
+    let local = args.local;
     let bot_args = config.merge(args);
 
-    info!("Starting PokeBot!");
-    debug!("Received CLI arguments: {:?}", std::env::args());
+    info!(root_logger, "Starting PokeBot!");
+    debug!(root_logger, "Received CLI arguments"; "args" => ?std::env::args());
 
-    if bot_args.local {
+    if local {
         let name = bot_args.names[0].clone();
-        let id = bot_args.ids.expect("identies should exists")[0].clone();
-
-        let disconnect_cb = Box::new(move |_, _, _| {});
+        let identity = bot_args.ids.expect("identies should exists")[0].clone();
 
         let bot_args = MusicBotArgs {
             name,
-            name_index: 0,
-            id_index: 0,
+            master: None,
             local: true,
             address: bot_args.address.clone(),
-            id,
+            identity,
             channel: String::from("local"),
             verbose: bot_args.verbose,
-            disconnect_cb,
+            logger: root_logger,
         };
-        MusicBot::new(bot_args).await.1.await;
+        MusicBot::new(bot_args).await;
+
+        ctrl_c.await??;
     } else {
         let domain = bot_args.domain.clone();
         let bind_address = bot_args.bind_address.clone();
-        let (bot, fut) = MasterBot::new(bot_args).await;
+        let bot_name = bot_args.master_name.clone();
+        let bot_logger = root_logger.new(o!("master" => bot_name.clone()));
+        let bot = MasterBot::new(bot_args, bot_logger).await;
+
+        let weak_bot = bot.downgrade();
 
-        thread::spawn(|| {
+        let clogger = root_logger.clone();
```

Comment:
lol

Reply:
It is gone after creating the `spawn_web_server` function.